### ISSUE TITLE
Missing support for modern transmission maps

### ISSUE THREAD

**Issue by AntonioCaroppos** [Unity-Technologies/ScriptableRenderPipeline#1689]
##### 2018.2.0b11 Windows

**Comment by sebastienlagarde** [comment]
TL;DR

Thickness map is (most of the time) the opposite of what other engine may call transmission map. i.e thicknessmap = 1 - transmission map.

>I want to use Transmission maps (which are physically correct)
Can you explain what is a 'physically correct' transmission map?
Transmission is name that we give to forward scattering inside a participating media when a light is behind the object, but in itself it is just scattering inside a volume with absorption and scattering parameters. And this depends on the length traveled in the volume. This length is approximated most of the time by a 'transmission map' i.e where the surface is thin, we tag the transmission map to say there is more backlight that are visible.  However as the goal is to represent a length information, what we expect to provide is not a 'transmission factor' but a thickness. So thickness map is the correct map to use here. In HDRP as we encode thickness on a R8, we only encode 0 to 1 and ask artists to provide a remap to real length in the UI. Transmission map is usually the opposite of this map (a bit like smoothness and roughness).

Regarding 'physically correct', the map that we provide (either transmission or thickness) is just an approximation anyway, as to get the correct length traveled inside the volume, we need to consider the point of view. So using a fixed map is wrong, and we should rather render back face then front face of a mesh to get a better approximation (however as it is more expensive we don't do it).

So there is no physically correct transmission map, there is no 'modern' transmission map (a modern way to do this would be to use ray tracing) and there is no 'old' thickness map. It is just a convention and we have chose a name that define the purpose. Thickness is about length traveled by the light in this medium.

**Comment by AntonioCaroppos** [comment]
Could you read here? I'm not sure who is correct. The shader masters (you) or the creator of Knald (the master of maps).

https://polycount.com/discussion/128895/knald-master-thread/p15

The author writes:

While there are always improvements to be made with any technique, we do consider our Transmission to produce the most realistic result. The generic Thickness maps that many solutions have implemented are fundamentally flawed and inaccurate when it comes to the transmission of light

Here is another example using the Hunter mesh (Art courtesy of Warren Boonzaaier) so you can see judge our Transmission solution directly with the alternative Thickness. In the Non-Knald thickness you can see there are major problems in multiple areas such as the eyelashes, eyeballs, the belts and the hood etc. While Thickness generated this way does a good job of generating local thickness it does not represent the true thickness of the entire mesh rendered as a coherent volume and is only suitable for singe objects like crystals or candles etc. when viewed in isolation.

**Comment by metalliandy** [comment]
You can simply invert the Transmission map from Knald and then use it as a direct replacement for "Thickness" inside Unity. You may want to play with the settings within Knald in order to get a similar intensity to what is typical with "Thickness" maps (changing the Attenuation to Ramp Linear and bumping the Scale slider to 1000% is usually pretty good).

Sebastian is correct in that having a baked map cannot produce a true physically based solution as the data is fixed in the form of a texture and to be absolutely physically correct this would need to be done in realtime.

With that said, we do feel that Knald produces the most realistic result when baking Transmission maps because it takes relative light transmission (energy) passing through the volume of the entire mesh into account (treating it as a coherent volume) when generating the texture, rather than simply generating local thickness. 

While not physically correct, the results speak for themselves :)

**Comment by AntonioCaroppos** [comment]
Why doesn't the HDRP Lit Shader accept Transmission maps right away? Or why do you (Andy) not create Thickness maps right away?
